tests/lib/k8s/caasp.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) 2020 SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# This module should take care of deploying kubernetes. There will likely be
+# multiple variations of an abstract base class to do so. However, the
+# implementation may need to require a certain base OS. For example, skuba
+# would require SLE and can raise an exception if that isn't provided.
+
+import logging
+import os
+import subprocess
+from tests.lib.k8s.kubernetes_base import KubernetesBase
+
+
+logger = logging.getLogger(__name__)
+
+
+class CaaSP(KubernetesBase):
+    def __init__(self, hardware):
+        super().__init__(hardware)
+        self.clusterpath = os.path.join(hardware.working_dir, 'cluster')
+        self._kubeconfig = os.path.join(self.hardware.working_dir,
+                                        'admin.conf')
+        self.ssh_agent()
+
+    def ssh_agent(self):
+        try:
+            res = subprocess.run(['ssh-agent'], check=True,
+                                 capture_output=True)
+        except subprocess.CalledProcessError:
+            msg = 'Failed to start ssh agent'
+            logger.error(msg)
```

Comment:
can we log the original error too? So we don't lose important debugging info?

Reply:
Done